src/autoscaler/db/sqldb/instancemetrics_sqldb.go

```diff
@@ -110,6 +110,9 @@ func (idb *InstanceMetricsSQLDB) SaveMetricsInBulk(metrics []*models.AppInstance
 	case "mysql":
 		sqlStr :="INSERT INTO appinstancemetrics(appid, instanceindex, collectedat, name, unit, value, timestamp)VALUES"
 		vals := []interface{}{}
+		if metrics == nil || len(metrics) == 0 {
```

Comment:
A transaction has created in https://github.com/cloudfoundry/app-autoscaler/blob/47490b6485063d6e1eb2f33db1f9fc19dc5d15f8/src/autoscaler/db/sqldb/instancemetrics_sqldb.go#L75,
so here it should be commit or rollback.